src/components/Restoration/CorrectCompanyName.vue

```diff
@@ -50,36 +36,43 @@ export default class CorrectCompanyName extends Vue {
 
   /** Called when component is mounted. */
   mounted (): void {
+    // *** TODO: get name request approved name?
+    const legalName = this.nameRequest.legalName
     // Set the current company name to the form
-    if (this.getNameRequestLegalName) {
-      this.companyName = this.getNameRequestLegalName
+    if (legalName) {
+      this.textfield = legalName
     }
   }
 
   /** Watch for form submission and emit results. */
   @Watch('formType')
   private onSubmit (): void {
-    // this component should only see correct-name form type
+    // process only when current form type matches
     if (this.formType === NameChangeOptions.CORRECT_NAME) {
       // set the new company name
-      this.setNameRequest({
-        ...this.getNameRequest,
-        legalName: this.companyName
+      // *** TODO: update name request approved name?
+      this.emitNameRequest({
+        ...this.nameRequest,
+        legalName: this.textfield
```

Comment:
WIP -- the Name Request object actually doesn't have a `legalName` property. Instead, the approved name needs to be emitted to the parent to eventually be set in the store.